esphome/components/ld2450/ld2450.cpp

```diff
@@ -0,0 +1,827 @@
+        sr->publish_state(val);
+      }
+    }
+    // DISTANCE
+    sensor::Sensor *sd = this->move_distance_sensors_[index];
+    if (sd != nullptr) {
+      val = (uint16_t) sqrt(
+          pow(this->decode_coordinate_(buffer[TARGET_X + index * 8], buffer[(TARGET_X + index * 8) + 1]), 2) +
+          pow(this->decode_coordinate_(buffer[TARGET_Y + index * 8], buffer[(TARGET_Y + index * 8) + 1]), 2));
+      td = val;
+      if (val > 0)
+        target_count++;
+
+      if (sd->get_state() != val) {
+        sd->publish_state(val);
+      }
+    }
+    // ANGLE
+    angle = calculate_angle_(static_cast<float>(ty), static_cast<float>(td));
+    if (tx > 0) {
+      angle = angle * -1;
+    }
+    sensor::Sensor *sa = this->move_angle_sensors_[index];
+    if (sa != nullptr) {
+      if (sa->get_state() != angle) {
+        sa->publish_state(angle);
+      }
+    }
+#endif
+    // DIRECTION
+#ifdef USE_TEXT_SENSOR
+    direction = get_direction_(ts);
+    if (td == 0)
+      direction = "NA";
+    text_sensor::TextSensor *tsd = this->direction_text_sensors_[index];
+    if (tsd != nullptr) {
+      if (tsd->get_state() != direction) {
+        tsd->publish_state(direction);
+      }
+    }
+#endif
+
+    // Store target info for zone target count
+    this->target_info_[index].x = tx;
+    this->target_info_[index].y = ty;
+    this->target_info_[index].is_moving = is_moving;
+
+  }  // End loop thru targets
+
+#ifdef USE_SENSOR
+  // Loop thru zones
+  uint8_t zone_still_targets = 0;
+  uint8_t zone_moving_targets = 0;
+  uint8_t zone_all_targets = 0;
+  for (index = 0; index < MAX_ZONES; index++) {
+    // Publish Still Target Count in Zones
+    sensor::Sensor *szstc = this->zone_still_target_count_sensors_[index];
+    if (szstc != nullptr) {
+      zone_still_targets = this->count_targets_in_zone_(this->zone_config_[index], false);
+      if (szstc->get_state() != zone_still_targets) {
+        szstc->publish_state(zone_still_targets);
+      }
+    }
+    // Publish Moving Target Count in Zones
+    sensor::Sensor *szmtc = this->zone_moving_target_count_sensors_[index];
+    if (szmtc != nullptr) {
+      zone_moving_targets = this->count_targets_in_zone_(this->zone_config_[index], true);
+      if (szmtc->get_state() != zone_moving_targets) {
+        szmtc->publish_state(zone_moving_targets);
+      }
+    }
+
+    zone_all_targets = zone_still_targets + zone_moving_targets;
+
+    // Publish All Target Count in Zones
+    sensor::Sensor *sztc = this->zone_target_count_sensors_[index];
+    if (sztc != nullptr) {
+      if (sztc->get_state() != zone_all_targets) {
+        sztc->publish_state(zone_all_targets);
+      }
+    }
+
+  }  // End loop thru zones
+
+  still_target_count = target_count - moving_target_count;
+  // Target Count
+  if (this->target_count_sensor_ != nullptr) {
+    if (this->target_count_sensor_->get_state() != target_count) {
+      this->target_count_sensor_->publish_state(target_count);
+    }
+  }
```

Comment:
Data from external hardware should not be deduplicated here. This is what `filters` are for. Please adjust all code that does this in this file
```suggestion
  if (this->target_count_sensor_ != nullptr) {
    this->target_count_sensor_->publish_state(target_count);
  }
```